moveit_core/robot_trajectory/test/test_robot_trajectory.cpp

```diff
@@ -115,6 +115,13 @@ class RobotTrajectoryTestFixture : public testing::Test
     std::vector<double> trajectory_first_state_after_update;
     trajectory_first_waypoint_after_update->copyJointGroupPositions(arm_jmg_name_, trajectory_first_state_after_update);
     EXPECT_EQ(trajectory_first_state[0], trajectory_first_state_after_update[0]);
+
+    // Modify the first waypoint duration
+    double trajectory_first_duration_before_update = trajectory->getWayPointDurationFromPrevious(0);
+    trajectory->setWayPointDurationFromPrevious(0, trajectory_first_duration_before_update + 0.1);
+
+    // Check that the trajectory's first duration was updated
+    EXPECT_NE(trajectory->getWayPointDurationFromPrevious(0), trajectory_first_duration_before_update);
```

Comment:
Why would you compare with `NE` when you know the value it's actually supposed to be? :smiley:

Reply:
I'm just copying what was done at L249, for example. Very well, I'll update this.